src/documents/CoreAPI.md

```diff
@@ -0,0 +1,635 @@
+
+### fluid.isDOMNode(totest)
+
+* `totest {Any}`
+* Returns: `{Boolean}`
+
+Returns `true` if the supplied object is a DOM node.
+
+### fluid.isDOMish(totest)
+
+* `totest {Any}`
+* Returns: `{Boolean}`
+
+Returns `true` if the supplied object is a DOM node or a jQuery object.
+
+### fluid.isUncopyable(totest)
+
+* `totest {Any}`
+* Returns: `{Boolean}`
+
+Returns `true` if it would be unwise to copy the supplied object. The framework will not attempt to copy any material which
+fails this test. An object which passes `fluid.isPrimitive`, `fluid.isDOMish` or fails `fluid.isPlainObject` will pass this test.
+
+### fluid.isComponent(totest)
+
+* `totest {Any}`
+* Returns: `{Boolean}`
+
+Returns `true` if the supplied object is an Infusion component.
+
+### fluid.isDestroyed(component)
+
+* `component {Component}`
+* Returns: `{Boolean}`
+
+Returns `true` if the supplied Infusion component has been destroyed.
+
+### fluid.hasGrade(options, gradeName)
+
+* `options {Object}`
+* `gradeName {String}`
+* Returns: `{Boolean}`
+
+Returns `true` if the supplied options ([defaults](#fluid-defaults-gradename-options-)) are those of a component which has the supplied grade. These must be options which have
+already been through options merging, since the test is a simpleminded one that simply checks whether the `gradeNames` entry in the options contains the supplied `gradeName`.
+
+### fluid.componentHasGrade(component, gradeName)
+
+* `component {Component}`
+* `gradeName {String}`
+* Returns: `{Boolean}`
+
+Returns `true` if the supplied component has the supplied `gradeName` as a parent grade. This is a rapid check which uses the framework's internal data structures for the lookup.
+
+
+## Basic implementation atoms
+
+### fluid.identity(arg)
+
+* `arg {Any}`
+* Returns: `{Any}`
+
+### fluid.notImplemented()
+
+A function which raises a failure if executed. This function name can also be used as a placeholder in [Invokers](Invokers.md) and
+namespaced [Listener](InfusionEventSystem.md) to create abstract (noninstantiable) [Component Grades](ComponentGrades.md).
+
+### fluid.each(source, func)
+
+* `source {Array|Object}` A container to be iterated over (may be null)
+* `func {Function} (element {Any}, index {Number|String}) → Any`. This function will be invoked once on each member of `source` and will receive the member and its key (either an integer for an array source,
+or a string for an object source) 
+* Returns: none
+
+Iterates over a supplied array or hash, invoking a function once for each element. Similar to [`jQuery.each`](http://api.jquery.com/jquery.each/) only the
+arguments to `func` are the right way round and the function does not explode on nonvalues. `fluid.each` on `null` or `undefined` is a no-op.
+
+### fluid.invokeLater(func) 
+
+A standard utility to schedule the invocation of a function after the current function call stack returns. On browsers this defaults to `setTimeout(func, 1)` but in
```

Comment:
All descriptions above have the parameters described after the heading, followed by the textual explanation. Is there a reason for the switch here?